packages/by-name/qemu-tdx-static/0001-avoid-duplicate-definitions.patch

```diff
@@ -0,0 +1,336 @@
+From e7a2545908f90aae6db1550900ca400b897ad117 Mon Sep 17 00:00:00 2001
+From: Tom Dohrmann <erbse.13@gmx.de>
+Date: Fri, 20 Sep 2024 16:01:15 +0200
+Subject: [PATCH 1/2] avoid duplicate definitions
+
+Another library already defines crc32c and this causes linker errors.
+Rename the function in QEMU to avoid conflicts.
+> /nix/store/q7cd30gzrkq720riqfm1myrvygv9v516-x86_64-unknown-linux-musl-binutils-2.42/bin/x86_64-unknown-linux-musl-ld: /nix/store/bnkaijycj1svgfkglzzwbzr76my1b0xz-util-linux-minimal-static-x86_64-unknown-linux-musl-2.39.4-lib/lib/libblkid.a(libcommon_la-crc32c.o): in function `crc32c':
+> (.text+0x0): multiple definition of `crc32c'; libqemuutil.a.p/util_crc32c.c.o:/build/qemu-9.0.1/build/../util/crc32c.c:109: first defined here
+---
+ block/iscsi.c                      | 6 +++---
+ block/vhdx.c                       | 4 ++--
+ hw/net/net_rx_pkt.c                | 2 +-
+ include/qemu/crc32c.h              | 4 ++--
+ roms/u-boot/fs/btrfs/crypto/hash.c | 2 +-
+ roms/u-boot/fs/btrfs/crypto/hash.h | 2 +-
+ roms/u-boot/fs/btrfs/ctree.c       | 2 +-
+ roms/u-boot/fs/btrfs/ctree.h       | 2 +-
+ roms/u-boot/include/u-boot/crc.h   | 2 +-
+ roms/u-boot/lib/crc32c.c           | 2 +-
+ target/arm/helper.c                | 6 +++---
+ target/arm/helper.h                | 2 +-
+ target/arm/tcg/helper-a64.c        | 6 +++---
+ target/arm/tcg/translate-a64.c     | 8 ++++----
+ target/loongarch/helper.h          | 2 +-
+ target/loongarch/op_helper.c       | 2 +-
+ util/crc32c.c                      | 6 +++---
+ 17 files changed, 30 insertions(+), 30 deletions(-)
+
+diff --git a/block/iscsi.c b/block/iscsi.c
+index 2ff14b74..d9f6af01 100644
+--- a/block/iscsi.c
++++ b/block/iscsi.c
+@@ -1374,13 +1374,13 @@ static void apply_header_digest(struct iscsi_context *iscsi, QemuOpts *opts,
+     digest = qemu_opt_get(opts, "header-digest");
+     if (!digest) {
+         iscsi_set_header_digest(iscsi, ISCSI_HEADER_DIGEST_NONE_CRC32C);
+-    } else if (!strcmp(digest, "crc32c")) {
++    } else if (!strcmp(digest, "__crc32c")) {
+         iscsi_set_header_digest(iscsi, ISCSI_HEADER_DIGEST_CRC32C);
+     } else if (!strcmp(digest, "none")) {
+         iscsi_set_header_digest(iscsi, ISCSI_HEADER_DIGEST_NONE);
+-    } else if (!strcmp(digest, "crc32c-none")) {
++    } else if (!strcmp(digest, "__crc32c-none")) {
+         iscsi_set_header_digest(iscsi, ISCSI_HEADER_DIGEST_CRC32C_NONE);
+-    } else if (!strcmp(digest, "none-crc32c")) {
++    } else if (!strcmp(digest, "none-__crc32c")) {
+         iscsi_set_header_digest(iscsi, ISCSI_HEADER_DIGEST_NONE_CRC32C);
+     } else {
+         error_setg(errp, "Invalid header-digest setting : %s", digest);
```

Comment:
This is not related to the C function name clash, is it?

Reply:
It is. There's a linker error due to conflicting definitions.

Reply:
Oh, wait, no, of course not, you're right.